dragonfly/implementations/psyllid_provider.py

```diff
@@ -233,51 +238,167 @@ def get_number_of_streams(self, channel):
+        logger.info('Setting psyllid power snr threshold to {}'.format(threshold))
+
+
+    def _get_fmt_snr_high_threshold(self, channel='a'):
+        request = '.active-config.{}.fmt.threshold-power-snr-high'.format(str(self.channel_dict[channel]))
+        threshold = self.provider.get(self.queue_dict[channel]+request)['threshold-power-snr-high']
+        return float(threshold)
+
+
+    def _set_trigger_mode(self, mode_id, channel='a'):
+        request = '.active-config.{}.fmt.trigger-mode'.format(str(self.channel_dict[channel]))
+        self.provider.set(self.queue_dict[channel]+request, mode_id)
+        logger.info('Setting psyllid trigger mode to {} threshold trigger'.format(mode_id))
+
+
+    def _get_trigger_mode(self, channel='a'):
+        request = '.active-config.{}.fmt.trigger-mode'.format(str(self.channel_dict[channel]))
+        trigger_mode = self.provider.get(self.queue_dict[channel]+request)['trigger-mode']
+        return trigger_mode
+
+
+    def _set_n_triggers(self, n_triggers, channel='a'):
+        request = '.active-config.{}.eb.n-triggers'.format(str(self.channel_dict[channel]))
+        self.provider.set(self.queue_dict[channel]+request, n_triggers)
+        logger.info('Setting psyllid n-trigger/skip-tolerance to {}'.format(n_triggers))
+
+
+    def _get_n_triggers(self, channel='a'):
+        request = '.active-config.{}.eb.n-triggers'.format(str(self.channel_dict[channel]))
+        n_triggers = self.provider.get(self.queue_dict[channel]+request)['n-triggers']
+        return n_triggers
+
+
+    # tell psyllid to record a frequency mask, write it to a json file and prepare for triggered run
+    def make_trigger_mask(self, channel='a', filename='~/fmt_mask.json'):
+        if self.mode_dict[channel] != 'triggered':
+            logger.error('Psyllid instance is not in triggered mode')
+            raise core.exceptions.DriplineGenericDAQError("Psyllid instance is not in triggered mode")
 
-        logger.info('freq-only')
-        request = 'run-daq-cmd.ch'+str(self.channel_dict[channel])+'.tfrr.freq-only'
+        logger.info('Switch tf_roach_receiver to freq-only')
+        request = 'run-daq-cmd.{}.tfrr.freq-only'.format(str(self.channel_dict[channel]))
         result = self.provider.cmd(self.queue_dict[channel],request)
 
-        logger.info('update-mask')
-        request = 'run-daq-cmd.ch'+str(self.channel_dict[channel])+'.fmt.update-mask'
+        logger.info('Switch frequency_mask_trigger to update-mask')
+        request = 'run-daq-cmd.{}.fmt.update-mask'.format(str(self.channel_dict[channel]))
         result = self.provider.cmd(self.queue_dict[channel],request)
         time.sleep(1)
 
-        logger.info('run')
-        request = 'start-run'
-        result = self.provider.cmd(self.queue_dict[channel], request)
+        logger.info('Start short run to record mask')
+        self.start_run(channel ,1000, '/home/project8/fmt_update.egg')
```

Comment:
this is not the trigger mask. the trigger mask is stored here:

        timestr = time.strftime("%Y%m%d_%H%M%S")
        filename = '{}_frequency_mask_channel_{}_cf_{}.json'.format(timestr, self.channel_id, self.freq_dict[self.channel_id])
        path = os.path.join(self.mask_target_path, filename)

*mask_target_path* has to be set in the config file of *r2_channel_x_daq_interface*.

To record a mask psyllid has to start a run in frequency-only mode. The mask that is recorded is written to a json file. But currently psyllid cannot start a run without creating an egg file. This file is empty and has no use. Therefore I thought I give it a hardcoded destination where it will always be overwritten. Where do you want that to be?

Reply:
@nsoblath how hard would it be for psyllid to be changed to not create this egg file? If we're going to create a dummy file we should put it into a place where we know it will be cleaned up (and ideally do that ourselves). The default should probably in `/tmp/` and the path for it configurable, that way we can send it somewhere for dung beetle or hornet or something to clean up if we decide we need that.

Reply:
I added a feature to psyllid that lets one turn off the use of egg files by sending a dl `set` command to `psyllid.use-monarch`.  `false` will turn it off, and `true` will turn it back on.

If it's set to `false` and a writer attempts to write to a file, psyllid will crash.